### PR TITLE
Self Signed Certs all have hardcoded names, not allowing for multiple Client VPNs in a region

### PR DESCRIPTION
**Pull request by garrinmf**
## what
The certificate names are all hardcoded, not allowing modification via context.

## why
In order to have multiple Client VPNs in the same region, the keys stored in SSM need to be unique, the hardcoded approach doesn't allow this.

I really don't know best practices around backwards compatibility or naming in general, just that the way it is doesn't allow for specifying unique names for the stored SSM keys.   let me know if a different approach would work better.

## references
* Closes https://github.com/cloudposse/terraform-aws-ec2-client-vpn/pull/31
* Upstream fix to use id instead of name https://github.com/cloudposse/terraform-aws-ssm-tls-self-signed-cert/pull/10

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -21,7 +21,7 @@ module "self_signed_cert_ca" {
   source  = "cloudposse/ssm-tls-self-signed-cert/aws"
   version = "0.4.0"
 
-  name = "self-signed-cert-ca"
+  name = "${module.this.context.name}-self-signed-cert-ca"
```
Comment (nitrocode) [inline main.tf:24]: Nice catch. All of these should probably be using attributes instead.

```suggestion
  attributes = ["self", "signed", "cert", "ca"]
```

File: main.tf [hunk 2]
```diff
@@ -55,7 +55,7 @@ module "self_signed_cert_root" {
   source  = "cloudposse/ssm-tls-self-signed-cert/aws"
   version = "0.4.0"
 
-  name = "self-signed-cert-root"
+  name = "${module.this.context.name}-self-signed-cert-root"
```
Comment (nitrocode) [inline main.tf:58]:
```suggestion
  attributes = ["self", "signed", "cert", "root"]
```

File: main.tf [hunk 3]
```diff
@@ -90,7 +90,7 @@ module "self_signed_cert_server" {
   source  = "cloudposse/ssm-tls-self-signed-cert/aws"
   version = "0.4.0"
 
-  name = "self-signed-cert-server"
+  name = "${module.this.context.name}-self-signed-cert-server"
```
Comment (nitrocode) [inline main.tf:93]:
```suggestion
  attributes = ["self", "signed", "cert", "server"]
```

**Comment by nitrocode** [comment]
@garrinmf please review the terratest failure if you get a chance

**Comment by garrinmf** [comment]
@nitrocode It doesn't seem like the self signed cert module, https://github.com/cloudposse/terraform-aws-ssm-tls-self-signed-cert, knows how to look to the attributes.   So all the keys are writing to the same SSM parameter name, in the test case it's `example.pem` as example is the `module.this.name`.  I pulled it and manually tried it locally and if I don't specify context (with a name) the keyname is actually just `.pem`.  If I do specify context with a name they're named the same, not taking into account attributes.  Not sure if this needs fixed downstream or a different approach in this module
with_context
default
?

**Comment by r351574nc3** [comment]
Looks like terratests fail though

image

@garrinmf could you look at that?

**Comment by garrinmf** [comment]
@nitrocode  Is there another approach, besides the attributes, that works better than my original attempt but the downstream packages work with?

**Comment by jabcoder** [comment]
+1

**Review by nitrocode — APPROVED** [review]

**Pull request merged**